mk/tools.mk

```diff
@@ -318,3 +320,21 @@ breakpad_clean:
 	$(V1) [ ! -d "$(BREAKPAD_DIR)" ] || $(RM) -rf $(BREAKPAD_DIR)
 	@echo " CLEAN        $(BREAKPAD_DL_FILE)"
 	$(V1) $(RM) -f $(BREAKPAD_DL_FILE)
+
+.PHONY: lint
+lint:
+	@command -v astyle >/dev/null 2>&1 || { echo >&2 "Error: astyle is required for linting, but it's not installed. Aborting."; exit 1; }
+	@echo "Found astyle: $$(astyle --version)"
+	@if ! astyle --version | grep -q "$(LINTER_ASTYLE_VERSION)"; then \
+		echo >&2 "Error: astyle version $(LINTER_ASTYLE_VERSION) is required. Aborting."; \
+		exit 1; \
+	fi
+	@echo "Applying astyle formatting..."
+	@if [ -z "$(LINT_TARGETS)" ]; then \
+		echo "Linting all C and header files in $(SRC_DIR)..."; \
+		find $(SRC_DIR) -type f \( -name '*.c' -o -name '*.h' \) -exec astyle --options=.astylerc {} +; \
```

Comment:
i feel i'd prefer this option to accept individual files or folders, not only folders.
i.e. not use the `find` command, but rather maybe just accept `$@`.
request other devs' opinions.

Reply:
This is the default behaviour, if devs didn't supply any argument to `LINT_TARGETS`. If you provide LINT_TARGETS you would be able to lint specific files.
Do you suggest we remove the option for implicit execution ?

Reply:
lint should check that working directory is clean (`if git diff-index --quiet HEAD --; then`)